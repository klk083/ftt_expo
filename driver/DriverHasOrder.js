import React from 'react'
import {View, Text, StyleSheet, TouchableOpacity, Alert, TextInput, Modal  } from 'react-native'
import { RFPercentage } from "react-native-responsive-fontsize";

import { is_order_accomplished, order_was_canceled, order_is_accomplished, reason_for_the_cancellation } from '../common_files/Texts'

export default class DriverHasOrder extends React.Component {
    state = {
        isModalVisible: false
    }

    render() {
        return (
            <View style={styles.container}>
                <View style={styles.info_container}>
                    <Text style={styles.is_order_accomplished}>{is_order_accomplished}</Text>
                </View>
                <View style={styles.accomplished_buttonContainer}>
                    <TouchableOpacity onPress={() => this.props.navigation.navigate('driver Home')}>
                        <Text style={styles.accomplished_button}>{order_is_accomplished}</Text>
                    </TouchableOpacity>
                </View>
                <View style={styles.orderCancel_buttonContainer}>
                    <TouchableOpacity>
                        <Text
                            style={styles.orderCancel_button}
                            onPress={() => Alert.alert(
                                'Hva er grunnen til kanselleringen?',
                                '',
                                [
                                    {
                                        text: 'Kunden mÃ¸tte ikke opp',
                                        onPress: () => this.props.navigation.navigate('driver Home'),
                                    },
                                    {},
                                    {
                                        text: 'Noe annet',
                                        onPress: () => {this.setState({isModalVisible: true})},
                                        style: 'cancel',
                                    },
                                ],
                                {
                                    cancelable: false
                                },
                            )}
                            >{order_was_canceled}</Text>
                    </TouchableOpacity>
                </View>
                {this.state.isModalVisible &&
                <Modal
                    animationType="none"
                    transparent={true}
                    style={{backgroundColor:'blue'}}
                    presentationStyle={'overFullScreen'}
                    onRequestClose={() => {this.setState({isModalVisible: false})}}
                >
                    <View style={styles.modalContainer}>
                        <View style={styles.modalView}>
                            <Text style={styles.modalText}>{reason_for_the_cancellation}</Text>
                            <TextInput
                                style={styles.reason}
                                placeholder="Skriv her"
                                keyboardAppearance="default"
                                multiline={true}
                                maxLength={200}
                                autoFocus={true}
                                onChangeText={this.handleTlf}
                            />
                            <View style={styles.sendReasonButtonContainer}>
                                <TouchableOpacity>
                                    <Text style={styles.sendReasonButton}>Send</Text>
                                </TouchableOpacity>
                            </View>
                        </View>
                    </View>
                </Modal>
                }
            </View>
        )
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        alignItems: 'center',
    },
    info_container: {
        flex: 0.1,
    },
    is_order_accomplished: {
        fontSize: RFPercentage(7),
        marginTop: 50,
        marginBottom: 50,
        textAlign: 'center',
    },
    accomplished_buttonContainer: {
        flex: 0.2,
        alignItems: 'center',
        backgroundColor: 'darkseagreen',
        borderRadius: 15,
        paddingLeft: 20,
        paddingRight: 20,
        marginTop: 100,
        justifyContent: 'center',
    },
    accomplished_button: {
        textAlign: 'center',
        fontSize: RFPercentage(5),
    },
    orderCancel_buttonContainer: {
        flex: 0.7,
        paddingBottom: 20,
        justifyContent: 'flex-end',
        alignItems: 'center',
    },
    orderCancel_button: {
        textAlign: 'center',
        fontSize: RFPercentage(3),
        padding: 10,
        backgroundColor: 'firebrick',
        borderRadius: 15,
    },


    modalContainer: {
        flex: 1,
        justifyContent: "center",
        alignItems: "center",
        backgroundColor: 'rgba(52, 52, 52, 0.8)',
    },
    modalView: {
        margin: 10,
        backgroundColor: "white",
        borderRadius: 20,
        paddingTop: 20,
        paddingLeft: 30,
        paddingRight: 30,
        paddingBottom: 20,
        alignItems: "center",
        shadowColor: "#000",
        shadowOffset: {
            width: 0,
            height: 2
        },
        shadowOpacity: 0.25,
        shadowRadius: 3.84,
        elevation: 100
    },
    modalText: {
        alignItems: 'flex-start',
        fontSize: RFPercentage(3.5),
        textAlign: 'center'
    },
    reason: {
        marginTop: 5,
        color: '#3467eb',
        fontSize: RFPercentage(3.5),
        marginLeft: 30,
        marginRight: 30,
    },
    sendReasonButtonContainer:  {
        paddingTop: 30,
        justifyContent: 'flex-end'
    },
    sendReasonButton: {
        fontSize: RFPercentage(3.5),
        backgroundColor: 'dodgerblue',
        borderRadius: 15,
        paddingVertical: 5,
        paddingHorizontal: 10
    }
});
