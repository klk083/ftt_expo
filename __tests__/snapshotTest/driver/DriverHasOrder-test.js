import React from 'react'
import DriverHasOrder from '../../../driver/DriverHasOrder'

import renderer from 'react-test-renderer'

test('renders correctly', () => {
    const tree = renderer.create(<DriverHasOrder />).toJSON()
    expect(tree).toMatchSnapshot()
})
