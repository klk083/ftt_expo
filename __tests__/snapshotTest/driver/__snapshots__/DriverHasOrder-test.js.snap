// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "space-between",
        "padding": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 0.2,
          "justifyContent": "center",
          "padding": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "flex": 1,
            "fontSize": 73,
            "textAlign": "center",
          }
        }
      >
        Er oppdraget utført?
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 0.5,
          "justifyContent": "flex-start",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "darkseagreen",
            "borderRadius": 15,
            "flex": 0.4,
            "justifyContent": "center",
            "opacity": 1,
            "paddingHorizontal": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 67,
              "textAlign": "center",
            }
          }
        >
          Oppdraget utført
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 0.2,
          "justifyContent": "flex-end",
          "padding": 10,
        }
      }
    >
      <View
        accessible={true}
        focusable={false}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "firebrick",
            "borderRadius": 15,
            "flex": 0.3,
            "justifyContent": "flex-end",
            "opacity": 1,
            "padding": 10,
          }
        }
      >
        <Text
          onPress={[Function]}
          style={
            Object {
              "fontSize": 40,
              "textAlign": "center",
            }
          }
        >
          Turen ble kansellert
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
